src/ucp/core/ucp_request.inl

```diff
@@ -352,22 +352,36 @@ ucp_request_send_buffer_reg(ucp_request_t *req, ucp_md_map_t md_map,
                                   req->send.mem_type, req, uct_flags);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_request_send_buffer_reg_lane_check(ucp_request_t *req, ucp_lane_index_t lane,
+                                       ucp_md_map_t prev_md_map, unsigned uct_flags)
+{
+    ucp_md_map_t md_map;
+
+    if (ucp_ep_md_attr(req->send.ep,
+                       lane)->cap.flags & UCT_MD_FLAG_NEED_MEMH) {
+        ucs_assert(ucp_ep_md_attr(req->send.ep,
+                                  lane)->cap.flags & UCT_MD_FLAG_REG);
+        md_map = UCS_BIT(ucp_ep_md_index(req->send.ep, lane)) | prev_md_map;
+        return ucp_request_send_buffer_reg(req, md_map, uct_flags);
+    }
```

Comment:
how are we sure this will not be used in rndv flow where rkey is needed?

Reply:
we're using `ucp_*_reg_lane()` functions for registering UCT MEMH for AM/PUT/GET Zcopy
I'm looking for the filed in `ucp_request_t` structure that can be used to understand what kind of operation we are doing (registering MEMH or MEMH for RKEY)